### PR TITLE
EclecticIQv3integration

### PR DESCRIPTION
**Pull request by mikhva-eiq**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
New version of integration with EclecticIQ Intelligence Center. New commands and functionality.

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ilappe** [comment]
Thanks @mikhva-eiq for the contribution!
1. what are the relation between `EclecticIQIntelligenceCenterv3` and `EclecticIQv2`? 
2. please sign the CLA https://github.com/demisto/content/pull/30508#issuecomment-1784288143
3. please fix the workflow issues [lint & unit test](https://circleci.com/gh/demisto/content/726879), [validation](https://circleci.com/gh/demisto/content/726884)

i'm reviewing the files..

**Comment by melamedbn** [comment]
Hi @mikhva-eiq,

I hope this message finds you well. Firstly, I'd like to express our appreciation for your contribution to our content.

Regarding the indicator fields, it would be great if we could align with our best practices. It helps in ensuring consistency and ease of use for users. For instance:

- Replacing `EclecticIQ Entity Description` with `Description`
- Replacing `EclecticIQ Entity Title` with `Title`
etc.


These changes can streamline the user's experience, especially when they are searching or querying for indicators.

If there's a specific reason behind introducing these new fields, we'd love to understand more. Your insights are always valuable.

Thanks again for your collaboration and understanding. Look forward to your feedback!

Best regards,
Ben

File: Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py [hunk 1]
```diff
@@ -0,0 +1,2439 @@
...
+    eiq = EclecticIQ_api(baseurl=SERVER, eiq_api_version=API_VERSION, username="", password=PASSWORD, verify_ssl=USE_SSL)
+
+    LOG('Command being called is {}'.format(demisto.command()))
+    command_func = COMMANDS.get(demisto.command())
+    if command_func is not None:
+        command_func()
+
+except Exception as e:
+    return_error('Error has occurred in EclecticIQ integration: {}\n {}'.format(str(e)))
+'''
```
Comment (ilappe) [inline Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py:2439]: please use the command calls within the `try except` block

**Review by ilappe — CHANGES_REQUESTED** [review]
Thanks @mikhva-eiq, your contribution is very appreciated!
in addition of [this comment](https://github.com/demisto/content/pull/30508#issuecomment-1787124782)
please see my review comments

**Comment by ilappe** [comment]
Hi @mikhva-eiq  , we haven’t heard from you in a while.
Do you need any help with the pull request?

Please feel free to reach out to me here or on Slack.
Thanks again for contributing to our repo, hope to hear from you soon.

File: Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.yml
[the remark this answers is not in the document]
Comment (mikhva-eiq) [reply Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.yml:1]: fixed.

File: Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py [hunk 2]
```diff
@@ -0,0 +1,2439 @@
...
+
+    if (type(query_result) is dict) or (type(query_result) is list):
+        output_result = []
+        for entity in query_result:
+            record = []
+            record = entity
+
+            record["observables_output"] = str(entity.get("observables_list"))
+            record["relationships_output"] = str(entity.get("relationships_list"))
+            output_result.append(record)
```
[the remark this answers is not in the document]
Comment (mikhva-eiq) [reply Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py:1885]: fixed

File: Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py [hunk 3]
```diff
@@ -0,0 +1,2439 @@
...
+                           )
+            )
+
+    elif query_result is False:
+        human_readable_title = "No entities found in EclecticIQ Intelligence Center."
+        human_readable = tableToMarkdown(human_readable_title, "")
+
+        return_results(CommandResults(readable_output=human_readable
+                           )
+            )
```
[the remark this answers is not in the document]
Comment (mikhva-eiq) [reply Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py:1906]: fixed

File: Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py [hunk 4]
```diff
@@ -0,0 +1,2439 @@
...
+    observable_type = demisto.args().get('observable_type', "")
+    observable_maliciousness = demisto.args().get('observable_maliciousness', "")
+    sighting_title = demisto.args().get('sighting_title', "")
+    sighting_description = demisto.args().get('sighting_description', "")
+    sighting_confidence = demisto.args().get('sighting_confidence', "")
+    sighting_impact = demisto.args().get('sighting_impact', "")
+    sighting_tag = demisto.args().get('sighting_tag', "").split(",")
+
+    if sighting_tag == [""]:
+        del sighting_tag[:]
```
[the remark this answers is not in the document]
Comment (mikhva-eiq) [reply Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py:1944]: fixed

**Comment by mikhva-eiq** [comment]
@ilappe could you please take a look, there are some checks failed which I cannot get clearly, like secrets found in readme are examples or ip/uris, also that's a message related to tlp_color which I cannot get why it appears (the same in my local demisto-sdk)

also want to ask, can we go forward without test coverage? if it's a must what is test coverage expected? anyway it will delay to fix that

**Comment by mikhva-eiq** [comment]
Hi @melamedbn regarding indicator fields, reason to add is following - EclecticIQ platform is TIP which aggregates threat intel and provides consolidated result, in case of our customers we have use case when they want to enrich/ingest TI from EIQ but still to have capability to have description field from different TI integration in XSOAR, thatst a reason why there are additional fields added.

File: Packs/EclecticIQ/Author_image.png
Comment (ilappe) [inline Packs/EclecticIQ/Author_image.png:1]: please revert this change
logo should be 120x50

**Comment by ilappe** [comment]
Hi @mikhva-eiq 
- secret error - if it's not a secret, please add the url mentioned in the [error](https://app.circleci.com/pipelines/github/demisto/content/319900/workflows/dfaa05b6-807f-462c-bfdf-b1f3075ad92f/jobs/730358?invite=true#step-118-450_47) to the ignore file Packs/EclecticIQ/.secrets-ignore.
- `tlp` error - i made a small change to fix it.
- unit test (coverage) - we must have a `*_test.py` file in the integration directory with tests for each command appear in the integration. you can create it based on the [previous test file ](https://github.com/demisto/content/blob/master/Packs/EclecticIQ/Integrations/EclecticIQv2/EclecticIQv2_test.py) and coverage should be at least 70%.
- please note the [lint errors](https://app.circleci.com/pipelines/github/demisto/content/319900/workflows/dfaa05b6-807f-462c-bfdf-b1f3075ad92f/jobs/730361?invite=true#step-120-9074_48), there is description regarding the issue
thanks in advance

let me know if you need any help

**Review by melamedbn — APPROVED** [review]

**Comment by zed-eiq** [comment]
hello! i see that @melamedbn has approved this PR. Just to be clear: do you still need us to make those changes you've described @ilappe here https://github.com/demisto/content/pull/30508#issuecomment-1822272715 ? 🙏

**Comment by ilappe** [comment]
@zed-eiq 
> hello! i see that @melamedbn has approved this PR. Just to be clear: do you still need us to make those changes you've described @ilappe here [#30508 (comment)](https://github.com/demisto/content/pull/30508#issuecomment-1822272715) ? 🙏

yes, he approved his responsibility items (like fields, playbooks etc)
but my comment is still relevant
thanks

**Comment by mikhva-eiq** [comment]
Hi @ilappe could you provide some guidance regarding Mypy errors? for me makes no sense following:
- Name "datetime" already defined (possibly by an import)  [no-redef] // it's first import, why error?
- "Type[datetime]" has no attribute "datetime"  [attr-defined] // it has that attribute, why error?
some errors related to "demisto Common"
- Incompatible types in assignment (expression has type "URL", variable has type "IP")  [assignment] 
and also
-home/circleci/project/Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py:2068:51: error: "object" has no attribute "get"  [attr-defined]     ...           record["observable_classification"] = maliciousness_to_meta...

whats expected from my side to resolve that? or just exclude that from Mypy scope??

**Comment by ilappe** [comment]
Hi @mikhva-eiq 
i push few small fix regarding the mypy errors
please validate them.
regarding the error "error: "object" has no attribute "get" [attr-defined]"
it's due to `maliciousness_to_meta` have the "Unknown" key with empty `{}`

you can ignore them like i did

**Comment by ilappe** [comment]
Hi @mikhva-eiq
what's the status here ?
if you need any help, i will be happy to help..

**Comment by mikhva-eiq** [comment]
@ilappe what could happen with ci/cd on PANW side? I made a couple of commits but in the checks see errors related to demisto-sdk or something else but not the integration.

**Pull request merged**